tekton/pipeline_run.go

```diff
@@ -101,6 +103,23 @@ func (r *ReleasePipelineRun) WithReleaseLabels(releaseName, releaseNamespace str
 	return r
 }
 
+// WithApplicationSnapshot adds a param containing the ApplicationSnapshot as a json string
+// to the release PipelineRun.
+func (r *ReleasePipelineRun) WithApplicationSnapshot(snapshot *v1alpha1.ApplicationSnapshot) *ReleasePipelineRun {
+	snapshotString, err := json.Marshal(snapshot.Spec)
+	if err != nil {
+		return nil
```

Comment:
This is wrong. You should never return nil in a chain or it will break the code. If we need to send an error back, we can do it by adding code. In this particular case, it should always work as a CRD will always generate a valid JSON so I'd say that we can remove it.